app/controllers/application_controller/queries.rb

```diff
@@ -0,0 +1,437 @@
+      if query_needs_update?(args, result)
+        result = create_query(model, result.flavor, result.params.merge(args))
+      end
+    # If no query found, just create a default one.
+    else
+      result = create_query(model, :all, args)
+    end
+    result
+  end
+
+  def dealphabetize_q_param
+    params[:q].dealphabetize
+  rescue StandardError
+    nil
+  end
+
+  def query_exists(params)
+    return unless params && (query = Query.safe_find(params))
+
+    query
+  end
+
+  # Turn old query into a new query for given model,
+  # (re-using the old query if it's still correct),
+  # and returning nil if no new query can be found.
+  def find_new_query_for_model(model, old_query)
+    old_query_correct_for_model(model, old_query) ||
+      old_query_coercable_for_model(model, old_query) ||
+      outer_query_correct_or_coerceable_for_model(model, old_query) ||
+      nil
+  end
+
+  def old_query_correct_for_model(model, old_query)
+    old_query if !old_query || (old_query.model.to_s == model)
+  end
+
+  def old_query_coercable_for_model(model, old_query)
+    old_query.coerce(model)
+  end
+
+  def outer_query_correct_or_coerceable_for_model(model, old_query)
+    return unless (outer_query = old_query.outer)
+
+    if outer_query.model.to_s == model
+      outer_query
+    elsif (coerced_outer_query = outer_query.coerce(model))
+      coerced_outer_query
+    end
+  end
+
+  def save_updated_query(result)
+    result.increment_access_count
+    result.save
+  end
+
+  def query_needs_update?(new_args, query)
+    new_args.any? { |_arg, val| query.params[:arg] != val }
+  end
+
+  def invalid_q_param?
+    params && params[:q] &&
+      !QueryRecord.exists?(id: params[:q].dealphabetize)
+  end
+
+  # Create a new Query of the given flavor for the given model.  Pass it
+  # in all the args you would to Query#new. *NOTE*: Not all flavors are
+  # capable of supplying defaults for every argument.
+  def create_query(model_symbol, flavor = :all, args = {})
+    Query.lookup(model_symbol, flavor, args)
+  end
+
+  private ##########
+
+  def save_query_unless_bot(result)
+    return unless result && !browser.bot?
+
+    result.increment_access_count
+    result.save
+  end
+
+  # Create a new query by adding a bounding box to the given one.
+  def restrict_query_to_box(query)
+    return query if params[:north].blank?
+
+    model = query.model.to_s.to_sym
+    flavor = query.flavor
+    tweaked_params = query.params.merge(tweaked_bounding_box_params)
+    Query.lookup(model, flavor, tweaked_params)
+  end
+
+  def tweaked_bounding_box_params
+    {
+      north: tweak_up(params[:north], 0.001, 90),
+      south: tweak_down(params[:south], 0.001, -90),
+      east: tweak_up(params[:east], 0.001, 180),
+      west: tweak_down(params[:west], 0.001, -180)
+    }
+  end
+
+  def tweak_up(value, amount, max)
+    [max, value.to_f + amount].min
+  end
+
+  def tweak_down(value, amount, min)
+    [min, value.to_f - amount].max
+  end
+
```

Comment:
Feels like these belong somewhere else, maybe in Query or their own class?

Reply:
Agree it could be elsewhere. However, I think we will not need to tweak bounding boxes here when we switch to AR rather than Query. Our scopes include the tweak, IIRC, so it may be going away soon, along with a lot of this module.